support/@trustfractal/polkadot-utils/src/lib/watcher.ts

```diff
@@ -38,12 +38,12 @@ export class TxnWatcher {
         // unhandled case below.
       } else if (result.status.isInBlock) {
         this.status = 'InBlock';
-        this.onInBlock.callAll();
+        this.onInBlock.callAll(result.status.asInBlock.toHex());
       } else if (result.status.isFinalized) {
-        this.onInBlock.callIfUncalled();
+        this.onInBlock.callIfUncalled(result.status.asFinalized.toHex());
```

Comment:
Is `asFinalized` the same hash as `asInBlock`? Otherwise, we could call this with the incorrect InBlock hash.

Reply:
I am not sure, but when the status is `isFinalized` the `result.status.asInBlock` doesn't exists. That is why I assumed it should be the correct hash.

Reply:
If you log them both with the example script that should let you know if they're the same.